src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,196 @@
+  if (error === undefined) {
+    return { fieldErrors: undefined, hasErrors: false };
+  }
+
+  if (Array.isArray(error)) {
+    return {
+      fieldErrors: error,
+      hasErrors: true,
+    };
+  }
+
+  return { fieldErrors: [error], hasErrors: true };
+};
+
+function AutoComplete<Item extends ListItemKey = ListItemKey>(
+  props: AutoCompleteProps<Item>
+) {
+  const mergedProps = useMergedProps(props, defaultAutoCompleteProps);
+  const theme = useTheme();
+
+  const {
+    label,
+    innerRef,
+    error,
+    onChange,
+    onBlur,
+    onFocus,
+    disabled,
+    helpText,
+    itemToString: itemToStringFn,
+    placeholder,
+    itemsFooter: Footer,
+    listItem: ListItem,
+    hideLabel,
+    testId,
+    value,
+    optional,
+    __internal__open,
+    __internal__highlightedIndex,
+    __internal__loading,
+    __internal__focus,
+  } = mergedProps;
+
+  const { items, getItems, status } = useGetItems(props.getItems);
+
+  const auxId = useMemo(() => `auto-complete-aux-${nanoid()}`, []);
+  const { fieldErrors, hasErrors } = getFieldErrors(error);
+  const describedBy = helpText || hasErrors ? auxId : undefined;
+
+  const itemToString = (item: Item | null): string =>
+    itemToStringFn ? itemToStringFn?.(item) : item ? String(item) : "";
+
+  const {
+    isOpen,
+    getMenuProps,
+    getInputProps,
+    getItemProps,
+    getComboboxProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    inputValue,
+    selectItem,
+    closeMenu,
+  } = useCombobox<Item | null>({
+    items: items || [],
+    defaultSelectedItem: value,
+    onInputValueChange: (changed) => {
+      getItems(changed);
+    },
+    onSelectedItemChange: (changed) => {
+      onChange?.(changed.selectedItem);
+    },
+    itemToString: (item) =>
+      itemToStringFn ? itemToStringFn?.(item) : item ? String(item) : "",
+  });
+
+  const onClear = () => {
+    selectItem(null);
+    closeMenu();
+  };
+
+  const menuIsOpen = isOpen || (__internal__open ?? false);
+  const showClearIcon = (menuIsOpen && items.length > 0) || inputValue !== "";
+  const showMagnifier = !showClearIcon;
+  const loading = status === "LOADING" || __internal__loading;
+
+  const renderListItem = (record: Item) => {
+    if (ListItem) {
+      return <ListItem inputValue={inputValue} item={record} />;
+    }
+    return itemToString ? itemToString(record) : record;
+  };
+
+  return (
+    <Field
+      optional={optional}
+      disabled={disabled}
+      label={label}
+      hideLabel={hideLabel}
+      auxId={auxId}
+      helpText={helpText}
+      errors={fieldErrors}
+      testId={testId}
+    >
+      <div css={theme.autoComplete.getCSS({ targetElement: "container" })}>
+        <div {...getComboboxProps()}>
+          <InternalInput
+            {...getInputProps({ refKey: "innerRef", ref: innerRef })}
+            label={label}
+            describedBy={describedBy}
+            onFocus={onFocus}
+            onBlur={onBlur}
```

Comment:
The `onBlur` won't work properly as this will trigger validation as soon as the input is blurred which would include clicking the clear button and clicking an item which might flash the validation error.

I would recommend taking a look at the `useWrapperFocus` hook in oscar and it's use in `AutoAddress` component.